src/Algorithms/HungarianAlgorithm.cs

```diff
@@ -0,0 +1,387 @@
+                    costMatrix[s, t] = getCost(workers[s], jobs[t]);
+
+            return HungarianAlgorithm.Run(costMatrix);
+        }
+
+        /// <summary>
+        /// Evaluates a minimal cost matching on two-dimensional vectors using the least-squares metric.
+        /// </summary>
+        /// <param name="from">The source vectors.</param>
+        /// <param name="to">The destination vectors.</param>
+        /// <returns>The minimal cost matching based on the least-squares metric. A value of -1 means the source vector is not assigned.</returns>
+        public static int[] Run(Vector2[] from, Vector2[] to)
+        {
+            return HungarianAlgorithm.Run(from, to, (u, v) => (v - u).LengthSquared);
+        }
+
+        /// <summary>
+        /// Evaluates a minimal cost matching on two-dimensional vectors using the least-squares metric.
+        /// </summary>
+        /// <param name="from">The source vectors.</param>
+        /// <param name="to">The destination vectors.</param>
+        /// <returns>The minimal cost matching based on the least-squares metric. A value of -1 means the source vector is not assigned.</returns>
+        public static int[] Run(Vector3[] from, Vector3[] to)
+        {
+            return HungarianAlgorithm.Run(from, to, (u, v) => (v - u).LengthSquared);
+        }
+        #endregion
+
+        #region Implementation
+        #region Fields
+        private readonly float[,] costMatrix;
+        private readonly int rows, cols, dim;
+        private readonly float[] labelSources;
+        private readonly float[] labelDests;
+        private readonly int[] minSlackDestBySource;
+        private readonly float[] minSlackValueBySource;
+        private readonly int[] sourceMatches;
+        private readonly int[] destMatches;
+        private readonly int[] parentSourceByCommittedDest;
+        private readonly bool[] matchedSources;
+        #endregion
+
+        #region Initializaion
+        /// <summary>
+        /// Construct an instance of the algorithm.
+        /// </summary>
+        /// <param name="costMatrix">the cost matrix, where matrix[i][j] holds the cost of assigning worker i to job j, for all i, j.</param>
+        private HungarianAlgorithm(float[,] costMatrix)
+        {
+            this.costMatrix = costMatrix;
+            this.rows = this.costMatrix.GetLength(0);
+            this.cols = this.costMatrix.GetLength(1);
+            this.dim = System.Math.Max(this.rows, this.cols);
+            this.labelSources = new float[this.dim];
+            this.labelDests = new float[this.dim];
+            this.minSlackDestBySource = new int[this.dim];
+            this.minSlackValueBySource = new float[this.dim];
+            this.matchedSources = new bool[this.dim];
+            this.parentSourceByCommittedDest = new int[this.dim];
+            this.sourceMatches = new int[this.dim];
+            this.destMatches = new int[this.dim];
+
+            for (int i = 0; i < this.dim; i++)
```

Comment:
This loop is insignificant compared to the overall running time of the algorithm (O(n) vs O(n^3)). Furthermore this would make the usage of these arrays less intuitive and very prone to errors in future changes. Therefore I think there is no valid argument to change this.